ddtrace/contrib/internal/asgi/middleware.py

```diff
@@ -280,6 +390,122 @@ async def wrapped_send(message):
+
+                _dd.dm.inherited
+                _dd.dm.service
+                _dd.dm.resource
+                """
+
+                try:
+                    message = await receive()
+                    if (
+                        self.integration_config._trace_asgi_websocket_messages
+                        and message["type"] == "websocket.receive"
+                    ):
+                        if self.integration_config._websocket_messages_separate:
+                            span_context_manager = self.tracer.start_span(
+                                name="websocket.receive",
+                                service=span.service,
+                                resource=f"websocket {scope.get('path', '')}",
+                                span_type="websocket",
+                                child_of=None,
+                            )
+                        else:
+                            span_context_manager = self.tracer.trace(
+                                name="websocket.receive",
+                                service=span.service,
+                                resource=f"websocket {scope.get('path', '')}",
+                                span_type="websocket",
+                            )
+                        with span_context_manager as recv_span:
+                            recv_span.set_tag_str(COMPONENT, self.integration_config.integration_name)
+                            recv_span.set_tag_str(SPAN_KIND, SpanKind.CONSUMER)
+                            core.dispatch("asgi.websocket.receive", (message,))
+                            # the span should have link to http handshake span
+                            # the link should have attribute dd.kind set to executed_by
+                            recv_span.set_link(
+                                trace_id=span.trace_id, span_id=span.span_id, attributes={"dd.kind": "executed_by"}
+                            )
+
+                            if "text" in message:
+                                recv_span.set_tag_str("websocket.message.type", "text")
+                                recv_span.set_metric("websocket.message.length", len(message["text"].encode("utf-8")))
+                            elif "binary" in message:
+                                recv_span.set_tag_str("websocket.message.type", "binary")
+                                recv_span.set_metric("websocket.message.length", len(message["bytes"]))
+
+                            # since asgi is a high level framework, frames is always 1
+                            recv_span.set_metric("websocket.message.frames", 1)
+
+                            if (
+                                self.integration_config._websocket_messages_separate
+                                and self.integration_config._asgi_websockets_inherit_sampling
+                            ):
+                                recv_span.set_metric("_dd.dm.inherited", 1)
```

Comment:
There is also the need, in this case, to lock the decision maker of the recv_span to the one of the handshake. This translates in words to have the same `_dd.p.dm` tag. I see that's handled kind of here (https://github.com/DataDog/dd-trace-py/blob/12a78bf1cf53d49c3603e6751521bfce38e7baae/ddtrace/internal/sampling.py#L270) 
Ideally we should set the same sampling mechanism and priority also in order not to have inconsistencies. The quick thing is just to set that meta like `recv_span._meta["_dd.p.dm"] = global_root_span._meta["_dd.p.dm"]`